cgroups.rst

```diff
@@ -1,151 +1,237 @@
+
+``--memory-swap`` sets the total amount of memory and swap space that a
+container can use. You must set ``--memory`` along with ``--memory-swap``. A
+value of ``-1`` means *unlimited swap*. If ``--memory-swap`` is not set or is
+``0``, then the container can use an amount of swap up to the value of
+``--memory``. It's easier to understand this flag with examples:
 
 .. code::
 
-   $ sudo {command} run --apply-cgroups path/to/cgroups.toml docker://alpine
+   # Run a container that can use up to 1G RAM, or swap if it is swapped out
+   $ {command} run --memory 1G myfirstapp.sif
+
+   # Run a container that can use up to 1G RAM, and no swap space
+   $ {command} run --memory 1G --memory-swap 1G myfirstapp.sif
+
+   # Run a container that can use up to 1G RAM, and unlimited swap space
+   $ {command} run --memory 1G --memory-swap -1 myfirstapp.sif
+
+   # Run a container that can use up to 1G RAM, and 1G swap space
+   $ {command} run --memory 1G --memory-swap 2G myfirstapp.sif
 
-After that, you can verify that the container is only using 500MB of
-memory. This example assumes that there is only one running container.
-If you are running multiple containers you will find multiple cgroups
-trees under the ``{command}`` directory.
+
+IO Limits
+=========
+
+.. note:: 
+
+   Requires the ``cfq`` or ``bfq`` IO scheduler to be configured for block IO on
+   the system. This is common on modern distributions, but not universal. Ask
+   your system administrator if IO limits are not working as expected.
+
+``--blkio-weight`` sets a relative weight for the container when performing
+block I/O, e.g. reading/writing to/from disk. The weight should be between 10
+and 1000, and will control how much I/O access a container recieves when there
+is contention for I/O with other containers. It may be useful to give high
+priority to a container that needs infrequent but time sensitive data access,
+running alongside an application that is continuously performing bulk reads.
+
+.. code:: 
+
+   # Container A - ten times as much block IO priority as container B
+   $ {command} run --blkio-weight 1000 myfirstapp.sif
+
+   # Container A - ten times less block IO priority as container A
+   $ {command} run --blkio-weight 100 mysecondapp.sif
+
+``--blkio-weight-device`` sets a relative weight for the container when performing
+block I/O on a specific device. Specify the device and weight as ``<device path>:weight``:
+
+.. code:: 
+
+   # Container A - ten times as much block IO priority as container B on disk /dev/sda
+   $ {command} run --blkio-weight-device /dev/sda:1000 myfirstapp.sif
+
+   # Container A - ten times less block IO priority as container A on disk /dev/sda
+   $ {command} run --blkio-weight-device /dev/sda:100 mysecondapp.sif
+
+********************************************
+ Applying Resource Limits From a TOML file
+********************************************
+
+{Project} 1.0.2 and above can directly apply resource limitations to systems
```

Comment:
Remove "1.0.2 and above".  In any case it would have been 1.1.0 but we don't document historical behavior in current documentation.